packages/eslint-plugin/src/rules/switch-exhaustiveness-check.ts

```diff
@@ -218,7 +238,9 @@ export default createRule<Options, MessageIds>({
     ): TSESLint.RuleFix {
       const lastCase =
         node.cases.length > 0 ? node.cases[node.cases.length - 1] : null;
-      const defaultCase = node.cases.find(caseEl => caseEl.test == null);
+      const defaultCase =
+        node.cases.find(caseEl => caseEl.test == null) ??
+        getCommentDefaultCase(node);
```

Comment:
[Refactor] This should have already been found by `getSwitchMetadata`. Now that the logic around finding a default case is getting more complex, let's take this in as a `fixSwitch` parameter instead of re-finding it.